lib/elasticsearch/index.rb

```diff
@@ -492,6 +493,32 @@ def prepare_popularity_field(doc_hash, popularities)
       doc_hash.merge("popularity" => pop)
     end
 
+    def prepare_mainstream_browse_page_field(doc_hash)
+      # Mainstream browse pages were modelled as three separate fields:
+      # section, subsection and subsubsection.  This is unhelpful in many ways,
+      # so model them instead as a single field containing the full path.
+      #
+      # In future, we'll get them in this form directly, at which point we'll
+      # also be able to there may be multiple browse pages tagged to a piece of
+      # content.
+      if doc_hash["mainstream_browse_pages"]
+        doc_hash
+      else
```

Comment:
The double-nested `if` gets a little harder to follow by the end.  This branch could be replaced by a guard condition:
`return doc_hash if doc_hash["mainstream_browse_pages"]`.